src/sd/host.rs

```diff
@@ -0,0 +1,169 @@
+                all(esp_idf_version_major = "5", esp_idf_version_minor = "1"),
+            )))]    // For ESP-IDF v5.2 and later
+            input_delay_phase: sdmmc_delay_phase_t_SDMMC_DELAY_PHASE_0,
+            #[cfg(not(any(
+                esp_idf_version_major = "4",
+                all(esp_idf_version_major = "5", esp_idf_version_minor = "0"),
+                all(esp_idf_version_major = "5", esp_idf_version_minor = "1"),
+            )))]   // For ESP-IDF v5.2 and later
+            set_input_delay: None,
+            command_timeout_ms: 0,
+        };
+
+        Self {
+            device: SdDevice::Spi(device),
+            host,
+        }
+    }
+
+    /// Create a new SD/MMC host with the default configuration.
+    /// This host will use the MMC slot 1, with 4-bit mode enabled, and max frequency set to 20MHz
+    #[cfg(esp_idf_soc_sdmmc_host_supported)]
+    pub fn new_with_mmc(configuration: SlotConfiguration<'d>) -> Self {
+        let host = sdmmc_host_t {
+            flags: _SDMMC_HOST_FLAG_8BIT
+                | _SDMMC_HOST_FLAG_4BIT
+                | _SDMMC_HOST_FLAG_1BIT
+                | _SDMMC_HOST_FLAG_DDR,
+            slot: configuration.get_slot() as i32,
+            max_freq_khz: _DEFAULT_FREQUENCY,
+            io_voltage: _DEFAULT_IO_VOLTAGE,
+            init: Some(sdmmc_host_init),
+            set_bus_width: Some(sdmmc_host_set_bus_width),
+            get_bus_width: Some(sdmmc_host_get_slot_width),
+            set_bus_ddr_mode: Some(sdmmc_host_set_bus_ddr_mode),
+            set_card_clk: Some(sdmmc_host_set_card_clk),
+            set_cclk_always_on: Some(sdmmc_host_set_cclk_always_on),
+            do_transaction: Some(sdmmc_host_do_transaction),
+            __bindgen_anon_1: sdmmc_host_t__bindgen_ty_1 {
+                deinit: Some(sdmmc_host_deinit),
+            },
+            io_int_enable: Some(sdmmc_host_io_int_enable),
+            io_int_wait: Some(sdmmc_host_io_int_wait),
+            get_real_freq: Some(sdmmc_host_get_real_freq),
+            #[cfg(not(any(
+                esp_idf_version_major = "4",
+                all(esp_idf_version_major = "5", esp_idf_version_minor = "0"),
+                all(esp_idf_version_major = "5", esp_idf_version_minor = "1"),
+             )))] // For ESP-IDF v5.2 and later            
+            input_delay_phase: sdmmc_delay_phase_t_SDMMC_DELAY_PHASE_0,
+            #[cfg(not(any(
+                esp_idf_version_major = "4",
+                all(esp_idf_version_major = "5", esp_idf_version_minor = "0"),
+                all(esp_idf_version_major = "5", esp_idf_version_minor = "1"),
+             )))] // For ESP-IDF v5.2 and later            
+            set_input_delay: Some(sdmmc_host_set_input_delay),
+            command_timeout_ms: 0,
+        };
+
+        Self {
+            device: SdDevice::Mmc(configuration),
+            host,
+        }
+    }
+
+    pub fn set_command_timeout(mut self, timeout: u32) -> Self {
```

Comment:
Belongs to a `Config` struct. See my other comments for the general pattern w.r.t. `Config`.